api/proto/teleport/legacy/types/types.proto

```diff
@@ -5830,3 +5830,101 @@ message MessageWithHeader {
     (gogoproto.embed) = true
   ];
 }
+
+// AWSMatcher matches AWS EC2 instances and AWS Databases
+message AWSMatcher {
```

Comment:
> Should we be defining these types in the same proto package as the DiscoveryConfigService that is also defined in RFD 125 instead of in types?

I created them in `types.proto` because I'm not seeing those as `DiscoveryConfigService` specific.
They should exist even if the Service doesn't.

( Maybe I'm not understanding the purpose of creating specific Services instead of adding everything into the AuthService/Types )

> Can we omit any of the gogo tags?

Yes, I'm pretty sure I can remove them all and use snake case for naming the fields.
This way, we should have the same case as is used for other yaml specs.
Is that what you are suggesting?

Reply:
I removed almost every gogo tag
The only ones that I didn't remove were the label matchers.
Labels use the `<string>:<string>` format, so we could use the proto definition `map<string, string>`

However, label matchers use `<string>:  List<string>`.
I'm not seeing any of the new services using this.
I see two options:
A) keep this as is:
```proto
  wrappers.LabelValues tags = 4 [
    (gogoproto.nullable) = false,
    (gogoproto.customtype) = "Labels"
  ];
```

B) define tags as:
```proto
message ListOfString {
  repeated string tags = 1;
}
...
  map<string, ListOfStrings> tags = 4;
```
But then we need a custom marshaller because it generates the following json/yaml
```json
"tags": {
  "x": { "tags": ["tag1", "tag2"] }
}
```

If we do this, we are basically back to the current `wrappers.LabelValues` with the gogo tag `custom type: Labels`

Reply:
No worries :+1: 
Thank you for the review
Let me know if there's anything else to discuss :pray: